react/frontpage/components/ShowsGraph.jsx

```diff
@@ -0,0 +1,215 @@
+// ShowsGraph.jsx
+
+var React = require('react');
+
+import Dates from '../../common/Dates';
+
+// styling
+require('./shows.scss');
+
+const week = ["sunday", "monday", "tuesday", "wednesday", "thursday", "friday", "saturday"];
+const dayWidth = `${100/8}%`;
+
+/*
+Full graph with schedule of shows
+
+@prop shows: list of show objects
+
+@prop currentShowID: id of current show
+@prop spotlightShowID: id of spotlight show
+@prop activeShowID: id of active show user clicked on
+@prop onShowClick(show): callback for when a show is made active
+*/
+const ShowsGraph = React.createClass({
+	getInitialState: function() {
+		return {};
+	},
+	componentWillMount() {
+		this.updateSchedule(this.props.shows);
+	},
+	componentWillReceiveProps(nextProps) {
+		if (nextProps.shows) {
+			this.updateSchedule(nextProps.shows);
+		}
+	},
+
+	updateSchedule: function(shows) {
+		var sorted = sortedShows(shows);
+		this.setState({
+			schedule: sorted
+		});
+	},
+	findStartEndTimes: function(start, end) {
+
+		for(start=0; start < 24; start ++){
+			for(var j = 0; j < this.props.shows.length; j++){
+				if(this.props.shows[j].time == Dates.availableTimes[start]){
+					break;
+				}
+			}
+		}
+
+		for(end = 23; end > 0; end--){
+			for(var j = 0; j < this.props.shows.length; j++){
+				if(this.props.shows[j].time == Dates.availableTimes[end]){
+					break;
+				}
+			}
+		}
+		start = 9;
+		end = 19;
+
+		return;
+	},
+
+	render: function() {
+
+		var dayTitles = week.map((day) => {
+			return (
+				<span className="headsStyle" style={{left: dayWidth, width: dayWidth}}>
+					{Dates.abbreviatedDay(day)}
+				</span> 
+			);
+		});
+
+		var showBlocks = [];
+
+		var fl1, fl2 = 0;
+
+		var start, end; 
+		for(start=1; start < 24; start ++){
+			for(var j = 0; j < this.props.shows.length; j++){
+				if(this.props.shows[j].time == Dates.availableTimes[start]){
+					fl1 = 1;
+					break;
+				}
+			}
+			if(fl1) {break;}
+		}
+
+		for(end = 23; end > 0; end--){
+			for(var j = 0; j < this.props.shows.length; j++){
+				if(this.props.shows[j].time == Dates.availableTimes[end]){
+					fl2 = 1;
+					break;
+				}
+			}
+			if(fl2) {break;}
+		}
```

Comment:
Can you comment what all this stuff does? It's supposed to limit times before and after? The variable names aren't very helpful for understanding